mail_tracking_mailgun/tests/test_mailgun.py

```diff
@@ -279,3 +309,50 @@ def test_event_dropped(self):
         self.assertEqual(event.error_type, reason)
         self.assertEqual(event.error_description, code)
         self.assertEqual(event.error_details, description)
+
+    @mock.patch(_packagepath + '.models.res_partner.requests')
+    def test_email_validity(self, mock_request):
+        self.partner.email_bounced = False
+        self.partner.email = 'info@tecnativa.com'
+        mock_request.get.return_value = {
+            'mailbox_verification': True,
+        }
+        self.partner.check_email_validity()
+        self.assertFalse(self.partner.email_bounced)
+        self.partner.email = 'xoxoxoxo@tecnativa.com'
+        mock_request.get.return_value = {
+            'mailbox_verification': False,
+        }
+        with self.assertRaises(UserError):
+            self.partner.check_email_validity()
+        self.assertTrue(self.partner.email_bounced)
+
+    @mock.patch(_packagepath + '.models.res_partner.requests')
+    def test_bounced(self, mock_request):
+        self.partner.email_bounced = True
+        mock_request.get.return_value.status_code = 404
+        self.partner.check_email_bounced()
+        self.assertFalse(self.partner.email_bounced)
+        mock_request.get.return_value.status_code = 200
+        self.partner.force_set_bounced()
+        self.partner.check_email_bounced()
+        self.assertTrue(self.partner.email_bounced)
+        mock_request.delete.return_value.status_code = 200
+        self.partner.force_unset_bounced()
+        self.assertFalse(self.partner.email_bounced)
+
+    def test_email_bounced_set(self):
+        message_number = len(self.partner.message_ids) + 1
+        self.partner._email_bounced_set('test_error', self.event)
+        self.assertEqual(len(self.partner.message_ids), message_number)
+        self.partner.email = ""
+        self.partner._email_bounced_set('test_error', self.event)
+        self.assertEqual(len(self.partner.message_ids), message_number)
+
+    @mock.patch(_packagepath + '.models.mail_tracking_email.requests')
+    def test_manual_check(self, mock_request):
+        mock_request.get.return_value = self.response
+        self.tracking_email.action_manual_check_mailgun()
+        event = self.env['mail.tracking.event'].search(
+            [('mailgun_id', '=', self.response['items'][0]['id'])])
+        self.assertEqual(event.event_type, self.respones['items'][0]['event'])
```

Comment:
It seems the [Travis failure](https://travis-ci.org/OCA/social/jobs/293783811#L1431) is here. It's "response".